Rtopel gradlew.ps1

bobtseattle opened the pull request:
Porting gradlew.bat to gradlew.ps1 to make running WPILib robot builds
easier on BSD student laptops.

bobtseattle committed:
Adding gradlew.ps1

Porting gradlew.bat to gradlew.ps1 to make running robot builds
easier on BSD student laptops.

bobtseattle committed:
Update comment

jamesdooley4 commented on gradlew.ps1:
```diff
@@ -0,0 +1,80 @@
+    Write-Output "location of your Java installation."
+    exit 1
+  }
+  $env:JAVA_EXE="$JAVA_EXE"
```
I don't see the environment variable used later, and this isn't set in the case where $env:JAVA_HOME is defined, so I'm pretty sure this can be deleted so the script doesn't add an unneeded environment variable when it's run

bobtseattle replied:
Yep

jamesdooley4 commented on gradlew.ps1:
```diff
@@ -0,0 +1,80 @@
+#
+# ##########################################################################
+
+$APP_BASE_NAME = $pwd
```
This returns the path for the current working directory, instead of the path the script is in (what ~dp0 refers to). I'd recommend using `Split-Path $MyInvocation.MyCommand.Path` instead, which should return the path of the script, so you can execute the gradlew.ps1 from any working directory & it will correctly start running from wherever the script is located.

jamesdooley4 replied:
In the original script this is also called APP_HOME, which I'd recommend using instead of APP_BASE_NAME (which is now what APP_NAME below represents).  That makes it more clear this is the "home" directory of the app.

bobtseattle replied:
Fixed.

jamesdooley4 commented on gradlew.ps1:
```diff
@@ -0,0 +1,80 @@
+$CLASSPATH = "$APP_BASE_NAME\gradle\wrapper\gradle-wrapper.jar"
+
+# Setup parameters
+$params = "$DEFAULT_JVM_OPTS"+" "+"$JAVA_OPTS"+" "+"$GRADLE_OPTS"+" "+"-Dorg.gradle.appname=$APP_NAME"+" "+"-classpath $CLASSPATH"+" "+"org.gradle.wrapper.GradleWrapperMain"+" "+"$CMD_LINE_ARGS"
```
I'd recommend keeping with JAVA_OPTS and GRADLE_OPTS being read from the environment instead of PowerShell variables, so use $env:JAVA_OPTS  and $env:GRADLE_OPTS instead.

bobtseattle replied:
Agreed.

bobtseattle committed:
Addressing feedback

Remove extranious setting.
Use APP_HOME instead of base name.
Make sure to get script location for APP_HOME.
